dhis-2/dhis-services/dhis-service-administration/src/main/resources/data-integrity-checks/programs/programs_inconsistent_tracked_entity_type.yaml

```diff
@@ -0,0 +1,108 @@
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+# Neither the name of the HISP project nor the names of its contributors may
+# be used to endorse or promote products derived from this software without
+# specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+# ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+---
+name: programs_inconsistent_tracked_entity_type
+description: Programs which are inconsistently linked to a tracked entity type.
+section: Programs
+section_order: 3
+summary_sql: >-
+  WITH rs as (
+  SELECT uid,name,
+  'Single event with tracked entity type' as comment
+  from program
+  where type = 'WITHOUT_REGISTRATION'
+  AND trackedentitytypeid IS NOT NULL
+  UNION
+  SELECT uid,name,
+  'Tracker program without tracked entity type' as comment
+  from program
+  where type = 'WITH_REGISTRATION'
+  AND trackedentitytypeid IS NULL
+  )
+  SELECT COUNT(*) as value,
+  100.0 * COUNT(*) / NULLIF( (SELECT COUNT(*) from program),0) as percent
+  from rs;
+details_sql: >-
+  SELECT uid,name,
+  'Single event with tracked entity type' as comment
+  from program
+  where type = 'WITHOUT_REGISTRATION'
+  AND trackedentitytypeid IS NOT NULL
+  UNION
+  SELECT uid,name,
+  'Tracker program without tracked entity type' as comment
+  from program
+  where type = 'WITH_REGISTRATION'
+  AND trackedentitytypeid IS NULL;
+severity: SEVERE
+introduction: >
+  Tracker programs, i.e. those with type 'WITH_REGISTRATION', 
+  should always have a tracked entity type associated with them.
+  Similarly, single event programs, i.e. those with type 'WITHOUT_REGISTRATION', 
+  should not have a tracked entity type associated with them.
+  
+  Tracker programs without a tracked entity type will not be able to register
+  new tracked entity instances and may cause errors when loading the Capture app.
+  Single event programs with a tracked entity type may cause confusion
+  for users as the tracked entity type is not used in single event programs.
+
+details_id_type: programs
+recommendation: >-
+  In order to fix tracker programs without a tracked entity type, the easiest solution would be to use
```

Comment:
Just to point out, it is also possible to attach a Tracked Entity Type to the problematic program using the (Old) Maintenance app. Maybe that is the "easiest" option for Tracker Programs atleast? API could be the second alternative. 

for Event Programs, the API would be needed yes.